### DISCUSSION TITLE
[Idea]  need of new modern webservice

### DISCUSSION THREAD

**Discussion by makhloughi** [PrestaShop/PrestaShop#35694] [Ideas]
### The problem

problem is non-well and non-useful formatted. actually there is no way to get public image url of a product. most useful fields are missing and for getting information about a product we need deep dive into multiple nested requests instead of having them in one object. this structure and thinking method belongs to under 2000!
what if we want to create a third-party program or mobile application for our website? is that right for viewing a product we make 4-5 or more requests to the server? also serve images not-optimized and with dynamic urls? where are using benefits of caching and webserver performance? we need a token to view images in third party app. is that safe and have good performance? absolutely not.
i studied PRs and issues. original developers like their old XML/JSON formatted API as it is and deny any changes to it and saying not to break old api. ok. then bring new webservice in a complete new endpoint and keep that junk in your code for backward compatibility. please do that, or say something useful, i don't think you have that!

### The solution

create a simple json api for getting products info at least. with complete structure and not to make 3-4 requests to get informations about a product ,images of it or attributes of it.
there is noooooooooooooo well documentation about this.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Amanek93** [comment]
I decided to create a mobile application based on a web service and unfortunately it was a difficult process.
I've been working on this for several months and every now and then I have doubts whether something is feasible. Even the simplest things cause problems. Due to the excess of requests, I am afraid that the application will not be efficient.


While I use the link_rewrite trick to download product photos, I have no idea how to download product attributes.

Does anyone have an idea on how to associate the id_product_attribute from the stock_availables resource with product_option_values?

In the case of combinations, I use the combinations resource, but I have no idea how to implement it in the case of simple products.

I need available sizes of clothes -> click on the size to select the appropriate id_product_attribute.

**Comment by silentyak** [comment]
According to the devs, the new API Platform is coming in v9:
https://github.com/PrestaShop/PrestaShop/discussions/34391#discussioncomment-7390398

It will have an OpenAPI spec, so you can (auto) generate the code bindings in whatever language you want.

**Comment by kpodemski** [comment]
Hello @makhloughi 

Web services are not ready by default to help you create a feature-rich mobile application, although we do have great documentation for them: https://devdocs.prestashop-project.org/8/webservice/.

How to extend it: https://devdocs.prestashop-project.org/8/modules/concepts/webservice/
We even have example Postman collections: https://devdocs.prestashop-project.org/8/webservice/tutorials/testing-webservice-postman/

If you want to avoid using web services and search for something better for your needs, you might want to look at https://docs.vuestorefront.io/prestashop/.

And, as @silentyak mentioned, with the upcoming release of version 9, PrestaShop's API is about to enter a new era, with API Platform-based solution :)

**Comment by makhloughi** [comment]
you are right that's extendable, but it's very resource draining to work with such API which requires to query the World to get a useful product object.
we are waiting for new version you mentioned!
